test/common/router/route_fuzz_test.cc

```diff
@@ -50,6 +64,19 @@ bool validateConfig(const test::common::router::RouteTestCase& input) {
       ENVOY_LOG_MISC(debug, "retry_policy_typed_config: not implemented");
       return false;
     }
+    if (virtual_host.has_matcher() && virtual_host.matcher().on_no_match().has_action() &&
+        virtual_host.matcher().on_no_match().action().name() ==
```

Comment:
Moving a discussion we had in another repo here:

This validates the `on_no_match` field of the matcher. However, there could be other fields (e.g., [FieldMatcher](https://github.com/envoyproxy/envoy/blob/f8f8628f4ec7cb2dcec92f094ffd4715dcc31dab/api/envoy/config/common/matcher/v3/matcher.proto#L99)) that will have the same issue.

I don't think there is a generic way to traverse this recursive data-structure, as it is very specific to how `Matcher` is defined. One can probably use protobuf reflection for this, but it may be an overkill.
I see 2 options here:
1. Handle the cases one-by-one as the fuzzer finds them.
2. Create a recursive function that traverses the Matcher object, and whenever it reaches an [`OnMatch` action](https://github.com/envoyproxy/envoy/blob/f8f8628f4ec7cb2dcec92f094ffd4715dcc31dab/api/envoy/config/common/matcher/v3/matcher.proto#L43) validates that it is not `filter_action` or `non_forwarding_action`.